vertx-web/src/main/java/io/vertx/ext/web/handler/impl/AuthenticationHandlerImpl.java

```diff
@@ -93,7 +94,7 @@ public void handle(RoutingContext ctx) {
             session.regenerateId();
           }
           // proceed with the router
-          ctx.next();
+          postAuthentication(ctx);
```

Comment:
If I'm not mistaken, on line 81 we also should call: `postAuthentication(ctx);` but I'm not sure as I'm reviewing using the github diff view, not really running the code ;-)

If you check it out I think we are ready to merge!

Reply:
Agree, and I think line 61 as well, which checks if there is already authenticated user or not.